src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1416,6 +1417,15 @@ private static void EmitTryMatchAtCurrentPosition(IndentedTextWriter writer, Reg
             HashSet<string> additionalDeclarations = new();
             Dictionary<string, string[]> additionalLocalFunctions = new();
 
+            // In debug builds, additional code is emitted to validate that the backtracking stack is being maintained appropriately.
```

Comment:
not present in interpreter either, I guess?

Reply:
right, but interpreter engine is very different to these other two. In the past, we've really tried to keep these two in sync given it is hard to debug the compiled engine, so it really helps to have them as close as possible since it helps debugging issues in the compiled engine using the sourcegen one. That said, with this PR we are adding logic to the sourcegen one that is not present in the compiled one, hence my comment.

Reply:
Even the comments in both files are in sync :)

All that said, now that CompileToAssembly is coming back (in debug at least) it may be easier to debug compiled engine.